migration/transitioning.rst

```diff
@@ -0,0 +1,126 @@
+
+Global and Reference Tables
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The tenant-specific tables do not require modification for use in Citus. During migration you would distribute them through the Citus cluster, sharding by tenant id. The other tables — admins and regions — do require modification.
+
+Consider the admins table first. It is aloof from tenant data, containing information about site administrators for the entire marketplace website, not individual stores. This table will not be involved in any SQL joins with the per-tenant tables. We call tables like these *global*. They usually relate to cross-tenant site operation or analytics. You can leave these tables unchanged, they will live on the Citus cluster master node and do not need to be distributed to worker nodes.
+
+The regions table is a different story. Because it lacks a tenant id but is involved in calculations of sales pricing along with per-tenant tables we call it a *reference* table.
+
+There are two ways to handle reference tables in the Citus multi-tenant use case:
+
+1. Duplicate
+2. Denormalize
+
+The first way is to add and a store_id column to regions, change the primary key from id alone to a composite key of id and store_id, and shard by the store_id column. (Citus requires primary keys and uniqueness constraints to contain the distribution column.)
+
+In addition to the schema change, you'll need to duplicate each row of the regions table, once for each store. To use the altered regions table in a join query, add a clause to match on store_id, for instance:
+
+.. code-block:: sql
+
+    -- Find adjusted purchase price with sales tax
+    -- for purchase number 1337
+
+    SELECT price + sales_tax
+      FROM purchases, regions
+     WHERE purchases.id = 1337
+       AND purchases.region_id = regions.id
+       -- include this condition:
+       AND purchases.store_id = regions.store_id
+
+This will run an efficient co-located join on the worker holding the shard for this store id.
+
+The second technique is to remove the regions table and denormalizing the database by embedding the sales_tax property directly into purchases. Duplicating just one property isn't too bad, but the situation isn't as nice for wider reference tables. Note that reference columns can also be stored in a single JSONB column in the denormalized table to prevent the latter from becoming awkwardly wide.
+
+Query Migration
+---------------
+
+To execute queries efficiently and isolate them within their tenant Citus needs to route them to a specific shard. Thus every query must identify which single tenant it involves. For non-joins this means that the *where* clause must filter by tenant id. In joins at least one of the tables must be filtered by tenant id. For instance:
```

Comment:
``In joins at least one of the tables must be filtered by tenant id.``. From speaking with @saicitus this isn't enough. With joins, either one table should be filtered by tenant-id and we add a join-clause between the two tables on tenant-id OR we need to filter both tables by tenant-id.

Sai's [presentation](https://docs.google.com/presentation/d/1HHIQSzEUscfzWcPWBGEmrANEqJEel5GR3IG65sVrvx8/edit#slide=id.g16a1d1e876_0_19) lists both options. Could you amend to include both options (with examples)?